yt/funcs.py

```diff
@@ -1191,14 +1191,13 @@ def validate_object(obj, data_type):
 
 def validate_axis(ds, axis):
     if ds is not None:
-        valid_axis = ds.coordinates.axis_name.keys()
+        valid_axis = sorted(
+            list(ds.coordinates.axis_name.keys()), key=lambda k: str(k).swapcase()
```

Comment:
What is this doing?

Reply:
sorting keys. When sorting strings, uppercase comes before lower case, so I'm reverting that order with a custom `lambda` key to match how the default list (`[0, 1, 2, "x", "y", "z", "X", "Y", "Z"]`) is ordered.

Reply:
Yes. Currently we rely on `pyupgrade` for this stuff, and it's documented as "timid" about string formatting. `ruff` is a little less so.